web/src/js/__tests__/ducks/modes/regularSpec.tsx

```diff
@@ -0,0 +1,123 @@
+import regularReducer, {
+    toggleRegular,
+    getMode,
+    initialState,
+    MODE_REGULAR_TOGGLE,
+} from "./../../../ducks/modes/regular";
+import * as options from "../../../ducks/options";
+import { TStore } from "../tutils";
+import fetchMock, { enableFetchMocks } from "jest-fetch-mock";
+
+describe("regularReducer", () => {
+    it("should return the initial state", () => {
+        const state = regularReducer(undefined, {});
+        expect(state).toEqual(initialState);
+    });
+
+    it("should dispatch MODE_REGULAR_TOGGLE and updateMode", async () => {
+        enableFetchMocks();
+
+        const store = TStore();
+
+        expect(store.getState().modes.regular.active).toBe(true);
+        await store.dispatch(toggleRegular());
+        expect(store.getState().modes.regular.active).toBe(false);
+        expect(fetchMock).toHaveBeenCalled();
+    });
+
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "regular", an host and a port', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["regular@http:8081"],
+                },
+            },
+        };
+        const newState = regularReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.listen_host).toBe("http");
+        expect(newState.listen_port).toBe(8081);
+    });
+
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "regular" and a port', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["regular@8081"],
+                },
+            },
+        };
+        const newState = regularReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.listen_host).toBe(undefined);
+        expect(newState.listen_port).toBe(8081);
+    });
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "regular"', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["regular"],
+                },
+            },
+        };
+        const newState = regularReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.listen_host).toBe(undefined);
+        expect(newState.listen_port).toBe(undefined);
+    });
```

Comment:
Same pattern here: These tests aren't too interesting because they hit exactly the same code path as the one above, but we're missing out on the interesting test that would uncover a bug (see below).